### ISSUE TITLE
atomic-reactor cannot push to insecure registry

### ISSUE THREAD

**Issue by ktdreyer** [containerbuildsystem/atomic-reactor#1283]
I am setting up a local OSBS environment. On a CentOS 7 VM, I'm running OpenShift 3.11 with `oc cluster up`, and I'm also running `docker-distribution` on port 5000 for a simple registry.

In my client `osbs.conf`, I have set `registry_uri = 192.168.122.68:5000`. This is an insecure registry, but I am able to "podman push" to it as long as I configure it as insecure in `/etc/containers/registries.conf`:

```
[registries.insecure]
registries = ['192.168.122.68']
```

When I run `osbs build` on my client, I see that atomic-reactor is unable to push to my configured registry. Here's the relevant atomic-reactor log:

```
2019-10-31 16:30:58,203 platform:- - atomic_reactor.plugin - DEBUG - running plugin 'tag_and_push'
2019-10-31 16:30:58,203 platform:- - atomic_reactor.plugin - INFO - running plugin instance with args: '{'registries': {'192.168.122.68:5000': {'insecure': True}}}'
2019-10-31 16:30:58,203 platform:- - atomic_reactor.plugins.tag_and_push - INFO - Registry 192.168.122.68:5000 secret None
2019-10-31 16:30:58,203 platform:- - atomic_reactor.core - INFO - tagging and pushing image 'sha256:efdbf57f40a7dbef72ec66ba6087fd732eb8dee9422789a4254ae7964bcf9de8' as '192.168.122.68:5000/myproject/hello-world:0.0.1-1'
2019-10-31 16:30:58,203 platform:- - atomic_reactor.core - DEBUG - image = 'sha256:efdbf57f40a7dbef72ec66ba6087fd732eb8dee9422789a4254ae7964bcf9de8', target_image = '192.168.122.68:5000/myproject/hello-world:0.0.1-1'
2019-10-31 16:30:58,203 platform:- - atomic_reactor.core - INFO - tagging image 'sha256:efdbf57f40a7dbef72ec66ba6087fd732eb8dee9422789a4254ae7964bcf9de8' as '192.168.122.68:5000/myproject/hello-world:0.0.1-1'
2019-10-31 16:30:58,203 platform:- - atomic_reactor.core - DEBUG - image = 'sha256:efdbf57f40a7dbef72ec66ba6087fd732eb8dee9422789a4254ae7964bcf9de8', target_image_name = '192.168.122.68:5000/myproject/hello-world:0.0.1-1'
2019-10-31 16:30:58,211 platform:- - atomic_reactor.core - INFO - pushing image '192.168.122.68:5000/myproject/hello-world:0.0.1-1'
2019-10-31 16:30:58,211 platform:- - atomic_reactor.core - DEBUG - image: '192.168.122.68:5000/myproject/hello-world:0.0.1-1', insecure: 'True'
2019-10-31 16:30:58,214 platform:- - atomic_reactor.util - INFO - wait_for_command
2019-10-31 16:30:58,218 platform:- - atomic_reactor.util - ERROR - {'errorDetail': {'message': 'Get https://192.168.122.68:5000/v1/_ping: http: server gave HTTP response to HTTPS client'}, 'error': 'Get https://192.168.122.68:5000/v1/_ping: http: server gave HTTP response to HTTPS client'}
```

I'm guessing that it is not enough to set this in my VM's `/etc/containers/registries.conf`, because `atomic-reactor` is still trying HTTPS here. How can I fix it?

**Comment by ktdreyer** [comment]
I've read https://github.com/containerbuildsystem/atomic-reactor/issues/372 , and I wasn't sure if that's the exact same issue.

**Comment by ktdreyer** [comment]
I found the problem. docker on CentOS 7 ignores `[registries.insecure]` in `/etc/containers/registries.conf` because it was reading the settings from `/etc/docker/daemon.json` instead. I don't know what created this json file (`oc cluster up` maybe?). Anyway it contained the following:
```json
{
 "insecure-registries": [
 "172.30.0.0/16"
 ]
}
```
I added `"192.168.122.0/24"` there and now atomic-reactor can push to the registry.
